aleph/tests/test_bookmarks_api.py

```diff
@@ -0,0 +1,113 @@
+import json
+from aleph.index.entities import index_entity
+from aleph.tests.util import TestCase, JSON
+from aleph.model import Bookmark
+from aleph.core import db
+
+
+class BookmarksApiTestCase(TestCase):
+    def setUp(self):
+        super(BookmarksApiTestCase, self).setUp()
+
+        self.role, self.headers = self.login()
+        self.collection = self.create_collection(self.role, label="Politicians")
+
+        data = {"schema": "Person", "properties": {"name": "Angela Merkel"}}
+        self.entity = self.create_entity(data=data, collection=self.collection)
+        index_entity(self.entity)
+
+    def test_bookmarks_index_auth(self):
+        res = self.client.get("/api/2/bookmarks")
+        assert res.status_code == 403, res
+
+        res = self.client.get("/api/2/bookmarks", headers=self.headers)
+        assert res.status_code == 200, res
+        assert res.json["total"] == 0, res.json
+
+    def test_bookmarks_index_results(self):
+        bookmark = Bookmark(entity_id=self.entity.id, role_id=self.role.id)
+        db.session.add(bookmark)
+        db.session.commit()
+
+        res = self.client.get("/api/2/bookmarks", headers=self.headers)
+        assert res.status_code == 200, res
+        assert res.json["total"] == 1, res.json
+        entity = res.json["results"][0]["entity"]
+        assert entity["properties"]["name"][0] == "Angela Merkel", res.json
+
+    def test_bookmarks_index_read_permission(self):
```

Comment:
This test is currently failing. This case is relevant if a user previously had access to a collection, bookmarked and entity from the collection, then lost access to the collection.

Right now, the bookmark is still included in the response, even though the actual entity data won’t be present:

```jsonc
{
  // ...
  "total": 1,
  "results": [
    {
      "created_at": "2022-01-31T...",
      // No `entity` key
    }
  ]
}
```

This is ok, although it would be a bit more elegant if the results simply excluded bookmarks that reference entities the user doesn’t have access to. Not sure though if it’s worth the effort to implement this.